### DISCUSSION TITLE
Input from system and not Windows

### DISCUSSION THREAD

**Discussion by Waxime64** [dotnet/Silk.NET#421] [Q&A] [answered]
I test your InputTest lab, but the mouse and keyboard are only detected in windows focus. How I can detect everywhere on the system ? Because i check for create service for add some effect everywhere on the system

**Comment by Perksey** [comment, marked as answer]
Hey, thanks for trying Silk.NET! (and our discussions system :P)

Currently, there is no way to do this within Silk.NET itself. I don't think it'll be possible with the current implementation due to limitations in the underlying windowing libraries we're using (GLFW and SDL), so a lot of rework would be required to do this which I unfortunately don't think will be coming any time in the near future.

Apologies for the inconvenience.

**Reply by Waxime64** [reply]
With Hook, it's work correctly
https://github.com/gmamaladze/globalmousekeyhook

**Reply by Perksey** [reply]
Awesome, recommend just using that as there's no way we can expose this behaviour in a non-platform-specific way.
